athena-gcs/src/test/java/com/amazonaws/athena/connectors/gcs/GcsMetadataHandlerTest.java

```diff
@@ -0,0 +1,325 @@
+        GetTableRequest getTableRequest = new GetTableRequest(federatedIdentity, QUERY_ID, "gcs", new TableName(SCHEMA_NAME, "testtable"));
+        Table table = new Table();
+        table.setName(TABLE_1);
+        table.setDatabaseName(DATABASE_NAME);
+        table.setParameters(ImmutableMap.of(CLASSIFICATION_GLUE_TABLE_PARAM, PARQUET));
+        table.setStorageDescriptor(new StorageDescriptor()
+                .withLocation(LOCATION).withColumns(new Column().withName("name").withType("String")));
+        table.setCatalogId(CATALOG);
+        List<Column> columns = List.of(
+                createColumn("name", "String")
+        );
+        table.setPartitionKeys(columns);
+        GetTableResult getTableResult = new GetTableResult();
+        getTableResult.setTable(table);
+        PowerMockito.when(awsGlue.getTable(any())).thenReturn(getTableResult);
+        StorageMetadata storageMetadata = mock(StorageMetadata.class);
+        Whitebox.setInternalState(gcsMetadataHandler, storageMetadata, storageMetadata);
+        PowerMockito.when(storageMetadata.buildTableSchema(any(), any())).thenReturn(schema);
+        GetTableResponse res = gcsMetadataHandler.doGetTable(blockAllocator, getTableRequest);
+        Field expectedField = res.getSchema().findField("name");
+        assertEquals(Types.MinorType.VARCHAR, Types.getMinorTypeForArrowType(expectedField.getType()));
+    }
+
+    @Test
+    public void testGetPartitions() throws Exception
+    {
+        Schema schema = SchemaBuilder.newBuilder().addField("id", new ArrowType.Int(64, false)).build();
+        Table table = new Table();
+        table.setName(TABLE_1);
+        table.setDatabaseName(DATABASE_NAME);
+        table.setParameters(ImmutableMap.of(CLASSIFICATION_GLUE_TABLE_PARAM, PARQUET,
+                PARTITION_PATTERN_KEY, "year={year}/birth_month{month}/{day}")
+        );
+        table.setStorageDescriptor(new StorageDescriptor()
+                .withLocation(LOCATION).withColumns(new Column()));
+        table.setCatalogId(CATALOG);
+        List<Column> columns = List.of(
+                createColumn("year", "varchar"),
+                createColumn("month", "varchar"),
+                createColumn("day", "varchar")
+        );
+        table.setPartitionKeys(columns);
+        GetTableResult getTableResult = new GetTableResult();
+        getTableResult.setTable(table);
+        PowerMockito.when(awsGlue.getTable(any())).thenReturn(getTableResult);
+        GetTableLayoutRequest getTableLayoutRequest = Mockito.mock(GetTableLayoutRequest.class);
+        Mockito.when(getTableLayoutRequest.getTableName()).thenReturn(new TableName(DATABASE_NAME, TABLE_1));
+        Mockito.when(getTableLayoutRequest.getCatalogName()).thenReturn(CATALOG_NAME);
+        Mockito.when(getTableLayoutRequest.getSchema()).thenReturn(schema);
+        Constraints constraints = new Constraints(createSummaryWithLValueRangeEqual("id", new ArrowType.Int(64, false), 1L));
```

Comment:
Is it intentional here to test constraints against the id column instead of the partition columns?

Note that I'm not saying that you need to change anything here, I'm just checking for what the intent of the test is.

Reply:
that was not intentional, I think it was incorrect. so corrected.